tests/test_features/test_jointplot.py

```diff
@@ -47,6 +48,13 @@
     pd = None
 
 
+def is_win_conda_env():
+    return (os.name == 'nt' and
+            os.path.exists(os.path.join(sys.prefix, 'conda-meta')))
+
+
+win_tol = 5.5 if is_win_conda_env() else None
```

Comment:
Yes, when I ran on conda on Linux most recently, there were 36 image comparisons failing. The extent of the differences previous to this week was something I discussed in https://github.com/DistrictDataLabs/yellowbrick/pull/823/#issuecomment-493755156

I would like to get the PR merged, as I'm sure you would! However, just before finalizing it, something occurred to me that I wanted to bring up. This might be something you already considered, or may want to discuss with others. It just became more clear in having to handle the multiple Windows builds, but it is more general than Appveyor/Windows comparisons.

Since part of the goal is to support conda across Windows and Linux (and I'm guessing Mac on Travis generally), I'm wondering if having a single `osenv` variable  might be the easiest way to handle the default tols for different OS/Python environment types combinations, as well as overrides. This is of course subjective, so it may be that for maintenance purposes and the contributor experience, a more coarse-grained OS-only set of tols are more appropriate.

Any approach has to allow maintainers to easily tune the default tols. There is the option of either just using an OS-based value for tol (`ostol`), or a more granular OS- and Python version-based tol (`osenvtol`).

If overrides were needed at the module or test level, an `osenv` variable could be imported for example (consistent with your suggestion to move the is_conda_env() out of the single module to base.py). This would allow for the maximum granularity without applying the same values across both regular Python and conda on the same OS.

**tol keyword arg(s)**: This is just my thought, but in either case, having just the single tol kwarg for overrides seems like it may require less effort to code in, and still work across OS platforms as well as Python environments.

Once there is an OS-specific tol, it becomes more effort to think "what is just plain tol for now? Is it best to handle conda separately, or just apply an OS-level tol, which may be higher than is needed for at least one of the Pythons (PyPI/conda)?" My current approach of using both the tol and windows_tol to handle both regular Python and conda having their own tols is definitely akward. The only good thing is that it applies more granularity to tols, based on the respective environment-specific builds.

If there is just one way to approach it by convention, whatever the convention, that can be applied to the PR.

The code that would allow for specifying default OS+py type tols would look something like this, if we wanted to allow for up to six default tols:

Within tests/base.py:

```

def is_conda_env():
	return os.path.exists(os.path.join(sys.prefix, 'conda-meta'))

def osenv_str():
	platform = {'win32': 'win', 'linux': 'lin', 'darwin'[:5]: 'mac'}
	py_or_conda = 'py' if not is_conda_env() else 'conda'
	return platform[sys.platform[:5]] + py_or_conda

# Configuration by maintainers

tol = 0.1

env_tols = {'linpy': tol, 'linconda': 0.1,
            'winpy': 0.1, 'winconda': 0.1,
             'macpy': 0.1, 'macconda': 0.1}

# Automatic based on configuration
os_tols = {'linux': max([v for k, v in env_tols.items() if 'lin' in k]),
            'win32': max([v for k, v in env_tols.items() if 'win' in k]),
            'darwin'[:5]: max([v for k, v in env_tols.items() if 'mac' in k])}


osenv = osenv_str()
osenvtol = env_tols[osenv]
ostol = os_tols[sys.platform[:5]]
```

For ImageComparisonclass, the `tol` could be specified by the maintainers like this:

`def __init__(tol=osenvtol)`

or like this:

`def __init__(tol=ostol)`

If there is no need for different OS/Python type tols in a module, an image comparison is the default
`self.assert_images_similar(visualizer)`

Or, if a test fails in a module despite the default tol from `__init__`, this is possible:

```
from tests.base import osenv, tol  # tol is 0.1, but no need to remember it

if osenv == 'winpy': tol = 0.5   # OR tol = 0.5 if osenv == 'winpy' else 0.1
self.assert_images_similar(visualizer, tol=tol)
```

Or, if there are 2 or more builds failing for a given module or test, something like this is possible at the top of a module or at the test level:

```
from tests.base import osenv, tol

# For example only
env_tols = {'winpy': 0.5, 'winconda': 5.5, 'linconda': 6.0}
if osenv in env_tols: tol = env_tols[osenv]
self.assert_images_similar(visualizer, tol=tol)
```

Reply:
@nickpowersys thanks for the thoughtful and detailed analysis on tolerances across operating systems. 

My sense is that this is a bigger conversation. In theory, having different tolerances across osenvs is suboptimal as we hope that the library could perform identically in each osenv. We probably should do a deeper dive into why the images produced are much different and draw inspiration from the approach that matplotlib takes to this very issue.

I recommend that you open up an [issue](https://github.com/DistrictDataLabs/yellowbrick/issues/new/choose) essentially with your proposal here. Perhaps you could list out the state of the image similarity tests across each of the osenvs to help steer the conversation and eventual work. 

For the moment thought, let's close out this PR knowing that we have windows conda and venv coverage in a much stronger place. It'll allow many other contributors to directly benefit from this work while the next task at hand is getting to the bottom of these bigger issues. 

What do you think about this approach?